src/win/winpty.cc

```diff
@@ -277,22 +300,33 @@ static NAN_METHOD(PtyKill) {
   if (info.Length() != 2 ||
       !info[0]->IsNumber() ||
       !info[1]->IsNumber()) {
-    Nan::ThrowError("Usage: pty.kill(pid, innerPidHandle)");
+    Nan::ThrowError("Usage: pty.kill(pid, innerPid)");
     return;
   }
 
-  int handle = info[0]->Int32Value(Nan::GetCurrentContext()).FromJust();
-  HANDLE innerPidHandle = (HANDLE)info[1]->Int32Value(Nan::GetCurrentContext()).FromJust();
+  DWORD pid = info[0]->Uint32Value(Nan::GetCurrentContext()).FromJust();
+  DWORD innerPid = info[1]->Uint32Value(Nan::GetCurrentContext()).FromJust();
 
-  winpty_t *pc = get_pipe_handle(handle);
+  winpty_t *pc = get_pipe_handle(pid);
   if (pc == nullptr) {
     Nan::ThrowError("Pty seems to have been killed already");
     return;
   }
 
-  assert(remove_pipe_handle(handle));
+  assert(remove_pipe_handle(pid));
```

Comment:
this seems incorrect, although not from the current change but the winpty agent cannot be closed here because it is needed for https://github.com/microsoft/node-pty/blob/14a2e96562367fb3e9fbfb4f5d634a128db129c5/src/windowsPtyAgent.ts#L182 which happens after call to kill. Maybe we need a separate internal API to kill the agent ? Thoughts @Tyriar

Reply:
Would collecting the process tree before calling `kill` fix the problem?

Reply:
Yup that is better approach.